pkg/types/imagebased/imagebased_config_types.go

```diff
@@ -0,0 +1,97 @@
+package imagebased
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	aiv1beta1 "github.com/openshift/assisted-service/api/v1beta1"
+	"github.com/openshift/installer/pkg/types"
+)
+
+// ImageBasedConfigVersion is the version supported by this package.
+const ImageBasedConfigVersion = "v1beta1"
+
+// InstallationConfig is the API for specifying configuration for the
+// image-based installation ISO.
+type InstallationConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// AdditionalTrustBundle is a PEM-encoded X.509 certificate bundle
+	// that will be added to the nodes' trusted certificate store.
+	//
+	// +optional
+	AdditionalTrustBundle string `json:"additionalTrustBundle,omitempty"`
+
+	// ExtraPartitionLabel label of extra partition used for /var/lib/containers.
+	// Default is varlibcontainers
+	// +optional
+	ExtraPartitionLabel string `json:"extraPartitionLabel,omitempty"`
+
+	// ExtraPartitionNumber number of extra partition used for /var/lib/containers.
+	// Default is 5
+	// +optional
+	ExtraPartitionNumber uint `json:"extraPartitionNumber,omitempty"`
+
+	// ExtraPartitionStart is the start of the /var/lib/containers partition. Free
+	// space before it will be allocated to system partition. It can be one of the
+	// following:
+	//   - Positive number: partition will start at position 120Gb of the disk and extend to the end of the disk. Example: 120Gb
+	//   - Negative number: partition will be of that precise size. Example: -40Gb
+	// It defaults to -40G (40GB before the end of the disk).
+	// +optional
+	ExtraPartitionStart string `json:"extraPartitionStart,omitempty"`
+
+	// ImageContentSources lists sources/repositories for the release-image content.
+	// The field is deprecated. Please use imageDigestSources.
+	// +optional
+	DeprecatedImageContentSources []types.ImageContentSource `json:"imageContentSources,omitempty"`
```

Comment:
Why we want to support DeprecatedImageContentSources ? Is this must? I mean it is already deprecated and we provide new api so why to add deprecated field?

Reply:
You are right, initially I thought that since the [install-config](https://github.com/openshift/installer/blob/master/pkg/types/installconfig.go#L144-L147) supports this, we should support it as well. But since this is a new API, i.e. we are actually not using the `install-config`, but our own config, we don't have to be backwards compatible, this field is redundant. I'll remove it, thanks for the validation :+1: